modular_skyrat/modules/borgs/code/robot_upgrade.dm

```diff
@@ -242,3 +170,268 @@
 		var/obj/item/inducer/cyborg/inducer = locate() in target_robot.model
 		if (inducer)
 			target_robot.model.remove_module(inducer, TRUE)
+
+/*
+*	ADVANCED MINING CYBORG UPGRADES
+*/
+
+/// Welder
+/obj/item/borg/upgrade/welder
+	name = "mining cyborg welder upgrade"
+	desc = "A normal welder with a larger tank for cyborgs."
+	icon_state = "cyborg_upgrade3"
+	require_model = TRUE
+	model_type = list(/obj/item/robot_model/miner)
+	model_flags = BORG_MODEL_MINER
+
+/obj/item/borg/upgrade/welder/action(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if(.)
+		for(var/obj/item/weldingtool/mini/W in R.model)
+			R.model.remove_module(W, TRUE)
+
+		var/obj/item/weldingtool/largetank/cyborg/WW = new /obj/item/weldingtool/largetank/cyborg(R.model)
+		R.model.basic_modules += WW
+		R.model.add_module(WW, FALSE, TRUE)
+
+/obj/item/borg/upgrade/welder/deactivate(mob/living/silicon/robot/R, user = usr)
+	. = ..()
+	if (.)
+		for(var/obj/item/weldingtool/largetank/cyborg/WW in R.model)
+			R.model.remove_module(WW, TRUE)
+
+		var/obj/item/weldingtool/mini/W = new (R.model)
+		R.model.basic_modules += W
+		R.model.add_module(W, FALSE, TRUE)
+
+/*
+*	ADVANCED CARGO CYBORG UPGRADES
+*/
+
+/// Better Clamp
+/obj/item/borg/hydraulic_clamp/better
+	name = "improved integrated hydraulic clamp"
+	desc = "A neat way to lift and move around crates for quick and painless deliveries!"
+	storage_capacity = 4
+	whitelisted_item_types = list(/obj/structure/closet/crate, /obj/item/delivery/big, /obj/item/delivery, /obj/item/bounty_cube) // If they want to carry a small package or a bounty cube instead, so be it, honestly.
+	whitelisted_item_description = "wrapped packages"
+	item_weight_limit = NONE
+	clamp_sound_volume = 50
+
+/obj/item/borg/hydraulic_clamp/better/examine(mob/user)
+	. = ..()
+	var/crate_count = contents.len
+	. += "There is currently <b>[crate_count > 0 ? crate_count : "no"]</b> crate[crate_count > 1 ? "s" : ""] stored in the clamp's internal storage."
+
+/obj/item/borg/hydraulic_clamp/mail
+	name = "integrated rapid mail delivery device"
+	desc = "Allows you to carry around a lot of mail, to distribute it around the station like the good little mailbot you are!"
+	icon = 'icons/obj/service/library.dmi'
+	icon_state = "bookbag"
+	storage_capacity = 100
+	loading_time = 0.25 SECONDS
+	unloading_time = 0.25 SECONDS
+	cooldown_duration = 0.25 SECONDS
+	whitelisted_item_types = list(/obj/item/mail)
+	whitelisted_item_description = "enveloppes"
```

Comment:
is this a typo

Reply:
PFFFFF, i didnt spellcheck anything when i was moving other code